util-cache/src/main/scala/sbt/util/SeparatedCache.scala

```diff
@@ -6,58 +6,59 @@ package sbt.util
 import scala.util.Try
 
 import sjsonnew.JsonFormat
+import sjsonnew.support.murmurhash.Hasher
 
 import CacheImplicits._
 
 /**
  * A cache that stores a single value.
  */
-trait SingletonCache[T] {
+trait SingletonCache[A] {
   /** Reads the cache from the backing `from`. */
-  def read(from: Input): T
+  def read(from: Input): A
 
   /** Writes `value` to the backing `to`. */
-  def write(to: Output, value: T): Unit
+  def write(to: Output, value: A): Unit
 
-  /** Equivalence for elements of type `T`. */
-  def equiv: Equiv[T]
+  /** Coded for elements of type `A`. */
+  def jsonFormat: JsonFormat[A]
```

Comment:
Can we remove this from the interface? At a glance of the diff I can't find evidence it's necessary.